Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -603,6 +605,35 @@ private extension ProductStore {
             completion(result)
         }
     }
+
+    func generateProductDetails(siteID: Int64, scannedTexts: [String], completion: @escaping (Result<ProductDetailsFromScannedTexts, Error>) -> Void) {
+        let prompt = [
+            "Write a name and description of a product for an online store given the array of scanned text strings from a packaging photo at the end.",
+            "Return only a JSON with the name in `name` field, description in `description` field, " +
+            "and the detected language as the locale identifier in `language` field.",
+            "The output should be in valid JSON format.",
+            "Identify the language detected in the array and use the same language to write the name and description.",
+            "Make the description 50-60 words or less.",
+            "Use a 9th grade reading level.",
+            "Perform in-depth keyword research relating to the product in the same language of the product title, " +
+            "and use them in your sentences without listing them out." +
+            "\(scannedTexts)"
+        ].joined(separator: "\n")
+        Task {
+            do {
+                let jsonString = try await generativeContentRemote.generateText(siteID: siteID, base: prompt, feature: .productDetailsFromScannedTexts)
+                guard let jsonData = jsonString.data(using: .utf8) else {
+                    return completion(.failure(DotcomError.resourceDoesNotExist))
+                }
+                let details = try JSONDecoder().decode(ProductDetailsFromScannedTextsResponse.self, from: jsonData)
+                await MainActor.run {
+                    completion(.success(.init(name: details.name, description: details.description, language: details.language)))
+                }
```

Comment:
Nit: as commented in https://github.com/woocommerce/woocommerce-ios/pull/10169#discussion_r1253977705 - I think we can get rid of the nested `await MainActor.run` and set `@MainActor` in the `Task` instead for simplicity.

Reply:
It was from following other actions, updated in https://github.com/woocommerce/woocommerce-ios/pull/10183/commits/eadcb26ef8ed6341c4887945e0b6adfb38073d6f